### ISSUE TITLE
`quotes`: updating avoidEscape to allow template literals to avoid escape sequences

### ISSUE THREAD

**Issue by sbdchd** [eslint-stylistic/eslint-stylistic#235]
### Clear and concise description of the problem

basing this off: https://github.com/eslint/eslint/issues/7660

Basically, with config: `["error", {avoidEscape: true}]`, template literals should be allowed when the other option is for the literal to have added escape sequences.

### Suggested solution

Instead of fixing:

```ts
x.startsWith(`"`) ? `"`  : null
```

into:

```ts
x.startsWith("\"") ? "\"" : null
```

it should be:

```ts
x.startsWith(`"`) ? `"` :  null
```

This would change the language from:

> "avoidEscape": true allows strings to use single-quotes or double-quotes so long as the string contains a quote that would have to be escaped otherwise

to:

> "avoidEscape": true allows strings to use single-quotes, double-quotes, or template literals so long as the string contains a quote that would have to be escaped otherwise

Further examples:

Instead of:

```diff
     const metaTags = document.querySelectorAll<HTMLMetaElement>(
-      `meta[property^="og:"]`,
+      "meta[property^=\"og:\"]",
     )
```

we'd leave it unchanged:

```ts
   const metaTags = document.querySelectorAll<HTMLMetaElement>(
     `meta[property^="og:"]`,
   )
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [X] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by antfu** [comment]
Making sense to me, PR welcome! Just to note that we might need to respect emcaVersion and avoid using it on versions that are not support template literals, otherwise I think it also make sense to enable it by default in modern JS.
